### PR TITLE
feat(note): add styled-system margin props - FE-3561

### PR DESCRIPTION
**Pull request by DipperTheDan**
### Proposed behaviour
Add styled-system margin props to the `Note` component

### Current behaviour
Styled-system is not available to the `Note` component

### Checklist

- [x] Commits follow our style guide
<del> - [ ] Screenshots are included in the PR if useful </del>
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
<del> - [ ] Cypress automation tests added or updated if required </del>
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
N/A

### Testing instructions
https://codesandbox.io/s/focused-tree-8c6h1

- apply the styled-system margin props to the Note component. Styles should behave as expected.

### CODE REVIEW

File: src/components/note/__internal__/note.component.js [hunk 1]
```diff
@@ -49,7 +57,7 @@ const Note = ({
   };
 
   return (
-    <StyledNote width={width} {...rest} data-component="note">
+    <StyledNote width={width} {...marginProps} {...rest} data-component="note">
```
Comment (edleeks87) [inline src/components/note/__internal__/note.component.js:60]: By spreading `rest` here you are automatically spreading the `marginProps`
```suggestion
    <StyledNote width={width} {...rest} data-component="note">
```

Comment (DipperTheDan) [reply src/components/note/__internal__/note.component.js]: Changed as requested. That should be sorted now 👍🏻

File: src/components/note/__internal__/note.component.js [hunk 2]
```diff
@@ -34,6 +40,8 @@ const Note = ({
     "<Note> inlineControl must be an instance of <ActionPopover>"
   );
 
+  filterStyledSystemMarginProps(rest);
```
[the remark this answers is not in the document]
Comment (DipperTheDan) [reply src/components/note/__internal__/note.component.js]: It's been removed 👍🏻

**Review by edleeks87 — DISMISSED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by KatarzynaQA — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**